File: tp1/TP01Q01 - Palíndromo em Java/palindromo.java
import java.io.*;

class palindromo{
	public static void main (String[] args){
		String frase = "";
		int i = 0, j = 0, tam = 0, resp = 1;
		
		while(i != 1){ //loop para leitura de todo arquivo
			frase = MyIO.readLine(); //criação do array frase
			
			if(frase.charAt(0) == 'F' && frase.charAt(1) == 'I' && frase.charAt(2) == 'M'){  //FIM presente no fim do arquivo
				i = 1;																			
			}else{
							
				tam = frase.length(); //tamanho do array frase
				
				j = tam - 1;
				
				for(int k = 0; k < j; k++, j--){				//leitura do array de tras (J--) pra frente (k++) 
                	if(frase.charAt(k) != frase.charAt(j)){		//efeturando comparações ate a metade do array
                    	resp = 0;
                    	k = j;
                	}
            	}
            	
            	if(resp == 1){ //condição para printar na tela se é palindromo ou não
                	MyIO.println("SIM");
            	}else{
                	MyIO.println("NAO");
            	}
            	resp = 1;
            	
			}
		}	
	}
}

File: tp1/TP01Q02 - Palíndromo em C/palindromo.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <locale.h>
#define TAM 1000 //definição de TAM para 1000 para facilitar ajustes no código

int main(){
    char frase[TAM];
    int tamanho = 0, i = 0, j = 0, k = 1;
    int resp = 1;
    
    scanf(" %[^\n]", frase); //leitura da primeira linha do array

	if(strcmp(frase, "FIM") == 0){ //comparação do array frase com FIM que corresponde à última linha do arquivo pub.in
        k = 0;
    }

    while(k != 0){	//loop para leitura das linhas do arquivo

 		tamanho = strlen(frase)-1; //tamanho do array

        j = tamanho;

        for(i = 0; i <= j/2; i++){	//leitura do array de tras (J--) pra frente (k++) efeturando comparações ate a metade do array
            if(frase[i] != frase[j-i]){
            	resp = 0;
               i = tamanho + 1;
            }
        }

        if(resp == 1){ //condição para printar se é palindromo ou não baseado nas comparações do for acima
            printf("SIM\n");
        }else{
            printf("NAO\n");
        }
        resp = 1;
       
        scanf(" %[^\n]", frase); //leitura de linha a linha do array
        
        if(strcmp(frase, "FIM") == 0){ //condição para ver se a frase lida corresponde ao final do arquivo identificado por FIM
            k = 0;
        }
        
    }
       
    return 0;
}
